app/models/UserAnswers.scala

```diff
@@ -73,7 +73,7 @@ final case class UserAnswers(
 
     updatedData.flatMap { d =>
       val updatedAnswers = copy(data = d)
-      page.cleanup(Some(value), updatedAnswers)
+      page.cleanup(Some(value), updatedAnswers, this)
```

Comment:
this?

Reply:
Yeah I don't like the "this", but we are inside the "old user answers" instance here and that is what we want to pass into this cleanup routine so we can do comparison between updated and old.